q2_mlab/db/maint.py

```diff
@@ -52,24 +56,46 @@ def add(engine: Engine, results: pd.DataFrame, parameters: dict,
     session.flush()
     params_id = params.id
 
-    # check if algorithm is valid
-    valid_algorithms = RegressionTask.algorithms
-    if algorithm not in valid_algorithms:
-        raise ValueError(f"Invalid choice '{algorithm}' for algorithm."
-                         f"Valid choices: {valid_algorithms}."
-                         )
+    # check if algorithm is valid, and of regression or classification
+    # and assign Score to the corresponding Regression or Classification table
+    valid_algorithms = set(RegressionTask.algorithms).union(
+        set(ClassificationTask.algorithms)
+    )
+    if algorithm in RegressionTask.algorithms:
+        Score = RegressionScore
+    elif algorithm in ClassificationTask.algorithms:
+        Score = ClassificationScore
+    else:
+        raise ValueError(
+            f"Invalid choice '{algorithm}' for algorithm."
+            f"Valid choices: {valid_algorithms}."
+        )
+
+    # check if uuid is in table, and skip it if so
+    # TODO once the code that pulls artifacts into the db and
+    # moves artifacts into an "inserted" directory is final, we
+    # may wish to change this to raise an error
```

Comment:
This check should be removed, as it is out of scope for the function. The function simply adds a record if told to do so. It would be somewhat limiting to automatically assume that no duplicate uuid should ever be added, and could also hide the case when other weird things happen.

There could be a separate convenience function for checking if the uuid exists, though.

Reply:
I think then we could do a separate method returning a boolean to check if uuids exists. Would this be more appropriate at the same level as `add_from_qza()` (so it would take the artifact and db_file and create an engine) or at the same level as `add()` (taking the uuid and engine, and could be called within `add_from_qza` as an option)

Reply:
I think the ideal solution is
1) leaving `add` as is
2) adding a function with a similar signature to `add`, called `check_uuid_exists`, that performs the `uuid` check
3) add an `add_duplicate_uuids=True` flag to `add_from_qza`, which controls whether the duplicate check is performed. We may consider changing the signature of `add_from_qza` to return a boolean that indicates whether the given artifact was actually inserted.